pandas/tests/resample/test_resampler_grouper.py

```diff
@@ -643,3 +643,30 @@ def test_groupby_resample_on_index_with_list_of_keys_missing_column():
     )
     with pytest.raises(KeyError, match="Columns not found"):
         df.groupby("group").resample("2D")[["val_not_in_dataframe"]].mean()
+
+
+def test_groupby_resample_agg_dict_works_for_as_index_false():
+    # GH 52397
+    expected = (
+        DataFrame(
+            {"a": np.repeat([0, 1, 2, 3, 4], 10), "b": range(50, 100)},
+            index=date_range(start=Timestamp.now(), freq="1min", periods=50),
+        )
+        .groupby("a", as_index=True)
+        .resample("2min")
+        .agg({"b": "min"})
+        .reset_index(drop=True)
+    )
+
+    result = (
+        DataFrame(
+            {"a": np.repeat([0, 1, 2, 3, 4], 10), "b": range(50, 100)},
+            index=date_range(start=Timestamp.now(), freq="1min", periods=50),
+        )
+        .groupby("a", as_index=False)
+        .resample("2min")
+        .agg({"b": "min"})
+        .reset_index(drop=True)
+    )
```

Comment:
In a test, don't modify result (especially a lossy modification). It is okay to modify expected after a computation to get it to match result.

Reply:
Thanks, updated